domain/consensus/datastructures/consensusstatestore/utxo.go

```diff
@@ -28,14 +28,14 @@ func (c *consensusStateStore) StageVirtualUTXODiff(virtualUTXODiff *model.UTXODi
 }
 
 func (c *consensusStateStore) commitVirtualUTXODiff(dbTx model.DBTransaction) error {
-	if c.stagedVirtualUTXOSet != nil {
-		return errors.New("cannot commit virtual UTXO diff while virtual UTXO set is staged")
-	}
-
 	if c.stagedVirtualUTXODiff == nil {
 		return nil
 	}
 
+	if c.stagedVirtualUTXOSet != nil {
```

Comment:
I'd rather if this check happens in the beginning of Commit.
Right now we check this twice (and the second one is redundant, because once we reach commitVirtualUTXOSet the UTXODiff has already been commited).

Even better - disallow staging if the other side is staged. This way we'll catch the error as it happens.